src/Implementation/IExecutable.cs

```diff
@@ -0,0 +1,9 @@
+﻿using System.Collections.Generic;
+
+namespace Symbolica.Implementation
+{
+    public interface IExecutable
```

Comment:
I don't think I agree with naming this `IExecutableState`, to me it seems to fall under this category of [code smell](https://degoes.net/articles/insufficiently-polymorphic). However, I may be missing something fundamental that means that this thing really is `IExecutableState`, but from what I see in the definition of this interface, nothing tells me it should be tied to `State` and so adding that to the name seems like an abstraction leak.